h2o-genmodel/src/main/java/hex/genmodel/easy/EasyPredictModelWrapper.java

```diff
@@ -568,6 +573,25 @@ public BinomialModelPrediction predictBinomial(RowData data, double offset) thro
     return p;
   }
 
+  /**
+   * Perform target encoding based on TargetEncoderMojoModel
+   * @param data RowData structure with data for which we want to produce transformations
+   * @return TargetEncoderPrediction with transformations ordered in accordance with corresponding categorical columns' indices in training data
+   * @throws PredictException
+   */
+  public TargetEncoderPrediction transformWithTargetEncoding(RowData data) throws PredictException{
+    if (! (m instanceof TargetEncoderMojoModel))
+      throw new PredictException("Model is not of the expected type, class = " + m.getClass().getSimpleName());
+
+    assert data.size() == m.nfeatures() : "There is a mismatch between number of features model was trained on (" + data.size() + ") and number of features in input RowData (" + m.nfeatures() + ")";
```

Comment:
SW still fails on this assert. (This [test](https://github.com/h2oai/sparkling-water/pull/1192/files#diff-7908001f8c9b38a0a2354f5996b51c6aR61) for instance)

The problem is if we want to encode only several columns (e.g. ["RACE", "DCAPS", "DPROPS"]) from the dataset, The `TargetEncoderMojoModel` considers all columns from the training dataset as features. (e.g. ["ID", "AGE", "RACE", "DPROPS", "DCAPS", "PSA",  "VAL", "GLEASON", "CAPSULE"] 

Maybe, the problem would be solved by setting `_names` on `TargetEncoderModel` to `TargetEncoderParameters._columnNamesToEncode`? WDYT?

Reply:
@mn-mikke maybe it would make more sense to consider as features only columns that we are trying to encode. But we need to be careful with indices....input for score0() method is just raw array. If our training frame contains all columns and test data contains only columns for encoding then we need to ensure that order in both cases is the same as we need to understand which element in array corresponds to what feature in training frame. I will try to prepare a test for this scenario.